pippo-core/src/main/java/ro/pippo/core/route/UrlResourceHandler.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2014 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ro.pippo.core.route;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ro.pippo.core.HttpConstants;
+import ro.pippo.core.PippoRuntimeException;
+import ro.pippo.core.util.StringUtils;
+
+import java.io.IOException;
+import java.net.URL;
+
+/**
+ * Serves static resources.
+ *
+ * @author James Moger
+ */
+public abstract class UrlResourceHandler extends ResourceHandler {
+
+    private static final Logger log = LoggerFactory.getLogger(UrlResourceHandler.class);
+
+    public UrlResourceHandler(String urlPath) {
+        super(urlPath);
+    }
+
+    @Override
+    public final void handleResource(String resourcePath, RouteContext routeContext) {
+        URL url = getResourceUrl(resourcePath);
+        if (url == null) {
+            routeContext.getResponse().notFound().commit();
+        } else {
+            streamResource(url, routeContext);
+        }
+    }
+
+    public abstract URL getResourceUrl(String resourcePath);
+
+    protected void streamResource(URL resourceUrl, RouteContext routeContext) {
```

Comment:
I can remove this method and put the content inline directly in the method `handleResource`.  I don't know if someone wants to bypass this part with caching.